### PR TITLE
Don't remove unique lock on shutdown

### PR DESCRIPTION
**Pull request by deltaroe**
@primiti

### CODE REVIEW

**Comment by deltaroe** [comment]
@HParker

File: spec/lib/middleware/server/unique_jobs_spec.rb [hunk 1]
```diff
@@ -8,21 +9,21 @@ module Server
           context "when class isn't a Sidekiq::Worker" do
             it 'returns false' do
               expect(subject.unlock_order_configured?(Class))
-                .to eq(false)
+                  .to eq(false)
```
Comment (HParker) [inline spec/lib/middleware/server/unique_jobs_spec.rb]: Why the change to 4 space indent here?

File: lib/sidekiq_unique_jobs/middleware/server/unique_jobs.rb [hunk 2]
```diff
@@ -15,8 +16,11 @@ def call(worker, item, _queue, redis_pool = nil)
           unlocked = before_yield? ? unlock(lock_key).inspect : 0
 
           yield
+        rescue Sidekiq::Shutdown
+          shutdown = true
+          raise
         ensure
-          if after_yield? || !defined? unlocked || unlocked != 1
+          if !shutdown && (after_yield? || !defined? unlocked || unlocked != 1)
```
Comment (HParker) [inline lib/sidekiq_unique_jobs/middleware/server/unique_jobs.rb:23]: What happens when shutdown is true and after_yield? is also true? shouldn't we unlock in that case?

File: spec/lib/middleware/server/unique_jobs_spec.rb (hunk 1 above)
Comment (deltaroe) [inline spec/lib/middleware/server/unique_jobs_spec.rb]: rubymine.  I'll go fix it

**Comment by HParker** [comment]
:+1:
